packages/replay-canvas/src/canvas.ts

```diff
@@ -0,0 +1,90 @@
+import { CanvasManager } from '@sentry-internal/rrweb';
+import type { CanvasManagerInterface } from '@sentry/replay';
+import type { Integration } from '@sentry/types';
+
+interface ReplayCanvasOptions {
+  quality: 'low' | 'medium' | 'high';
+}
+
+export interface ReplayCanvasIntegrationOptions {
+  recordCanvas: true;
+  getCanvasManager: (options: ConstructorParameters<typeof CanvasManager>[0]) => CanvasManagerInterface;
+  sampling: {
+    canvas: number;
+  };
+  dataURLOptions: {
+    type: string;
+    quality: number;
+  };
+}
+
+const CANVAS_QUALITY = {
+  low: {
+    sampling: {
+      canvas: 1,
+    },
+    dataURLOptions: {
+      type: 'image/webp',
+      quality: 0.25,
+    },
+  },
+  medium: {
+    sampling: {
+      canvas: 2,
+    },
+    dataURLOptions: {
+      type: 'image/webp',
+      quality: 0.4,
+    },
+  },
+  high: {
+    sampling: {
+      canvas: 4,
+    },
+    dataURLOptions: {
+      type: 'image/webp',
+      quality: 0.5,
+    },
+  },
+};
+
+/** An integration to add canvas recording to replay. */
+export class ReplayCanvas implements Integration {
```

Comment:
can we make this a functional component right away? We are migrating there for v8 and this will make it easier to update this then later. So bascially:

```ts
const INTEGRATION_NAME = 'ReplayCanvas';
const replayCanvasIntegration = ((options: Partial<ReplayCanvasOptions>) => {
  return {
    name: INTEGRATION_NAME,
    getOptions() {
      // ...
   }
}) satisfies IntegrationFn;

export const ReplayCanvasIntegration = convertIntegrationFnToClass(replayCanvasIntegration, INTEGRATION_NAME);